File: game/src/test/java/integration/cards/CardInstanceFactoryTest.java
package integration.cards;

import com.matag.cardinstance.CardInstance;
import com.matag.cardinstance.CardInstanceFactory;
import com.matag.cards.Cards;
import com.matag.game.status.GameStatus;
import integration.TestUtils;
import integration.TestUtilsConfiguration;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@ContextConfiguration(classes = TestUtilsConfiguration.class)
public class CardInstanceFactoryTest {

  @Autowired
  private CardInstanceFactory cardInstanceFactory;

  @Autowired
  private Cards cards;

  @Autowired
  private TestUtils testUtils;

  @Test
  public void shouldCreateACardInstance() {
    // Given
    GameStatus gameStatus = testUtils.testGameStatus();

    // When
    CardInstance cardInstance = cardInstanceFactory.create(gameStatus, 1, cards.get("Short Sword"), "player-name");

    // Then
    assertThat(cardInstance.getId()).isEqualTo(1);
    assertThat(cardInstance.getOwner()).isEqualTo("player-name");
  }

  @Test
  public void shouldCreateTwoDifferentCardInstances() {
    // Given
    GameStatus gameStatus = testUtils.testGameStatus();

    // When
    CardInstance cardInstance1 = cardInstanceFactory.create(gameStatus, 1, cards.get("Short Sword"), "player-name");
    CardInstance cardInstance2 = cardInstanceFactory.create(gameStatus, 2, cards.get("Befuddle"), "opponent-name");

    // Then
    assertThat(cardInstance1).isNotSameAs(cardInstance2);
  }

}

File: src/test/java/integration/cardinstance/CardInstanceFactoryTest.kt
package integration.cardinstance

import com.matag.cards.Cards
import com.matag.game.cardinstance.CardInstance
import com.matag.game.cardinstance.CardInstanceFactory
import integration.TestUtils
import integration.TestUtilsConfiguration
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [TestUtilsConfiguration::class])
class CardInstanceFactoryTest {
    @Autowired
    private val cardInstanceFactory: CardInstanceFactory? = null

    @Autowired
    private val cards: Cards? = null

    @Autowired
    private val testUtils: TestUtils? = null

    @Test
    fun shouldCreateACardInstance() {
        // Given
        val gameStatus = testUtils!!.testGameStatus()

        // When
        val cardInstance = cardInstanceFactory!!.create(gameStatus, 1, cards!!.get("Short Sword"), "player-name")

        // Then
        Assertions.assertThat(cardInstance.getId()).isEqualTo(1)
        Assertions.assertThat(cardInstance.getOwner()).isEqualTo("player-name")
    }

    @Test
    fun shouldCreateTwoDifferentCardInstances() {
        // Given
        val gameStatus = testUtils!!.testGameStatus()

        // When
        val cardInstance1 = cardInstanceFactory!!.create(gameStatus, 1, cards!!.get("Short Sword"), "player-name")
        val cardInstance2 = cardInstanceFactory.create(gameStatus, 2, cards.get("Befuddle"), "opponent-name")

        // Then
        Assertions.assertThat<CardInstance?>(cardInstance1).isNotSameAs(cardInstance2)
    }
}